eng/scripts/Create-APIView.ps1

```diff
@@ -86,6 +86,30 @@ function Get-ResourceProviderFromReadMePath {
     return $null
 }
 
+function Get-ImpactedTypespecProjects {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$TypeSpecFile
+    )
+    $projectRootPaths = [System.Collections.Generic.HashSet[string]]::new()
+    $filePathParts = $TypeSpecFile.split([IO.Path]::DirectorySeparatorChar)
+    $typeSpecProjectBaseDirectory = $filePathParts[0..($($filePathParts.Length)-2)] -join [IO.Path]::DirectorySeparatorChar
+    $configFilesInTypeSpecProjects = Get-ChildItem -Path $typeSpecProjectBaseDirectory -File "tspconfig.yaml" -Recurse
+    if ($configFilesInTypeSpecProjects) {
+      foreach($configFilesInTypeSpecProject in $configFilesInTypeSpecProjects) {
+        $entryPointFile = Get-ChildItem -Path $($configFilesInTypeSpecProject.Directory.FullName) -File "main.tsp" 
+        if ($entryPointFile) {
+          $projectRootPaths.Add($configFilesInTypeSpecProject.Directory.FullName) | Out-Null
```

Comment:
You don't need to loop again to check for main.tsp in current files path if you find tspconfig.yaml and main.tsp in same folder. A project should not have these two files in same directory in different places. So as soon as you detect both files a directory then you can return it as value.